lib/Sema/TypeCheckDecl.cpp

```diff
@@ -1171,6 +1173,98 @@ doesAccessorNeedDynamicAttribute(AccessorDecl *accessor, Evaluator &evaluator) {
   }
 }
 
+llvm::Expected<bool>
+IsFinalRequest::evaluate(Evaluator &evaluator, ValueDecl *decl) const {
+  switch (decl->getKind()) {
+    case DeclKind::Var: {
+      // Properties are final if they are declared 'static' or a 'let'
+      auto *VD = cast<VarDecl>(decl);
+      if (auto *nominalDecl = VD->getDeclContext()->getSelfNominalTypeDecl()) {
```

Comment:
You can replace getSelfNominalTypeDecl() with getSelfClassDecl(), and remove the `isa<ClassDecl>` check below. None of this code needs to run for anything that's not a class.